File: routes/activities.js
/* ═══════════════════════════════════════════
   Activity Routes — Task activity log
   ═══════════════════════════════════════════ */

const { pool } = require('../db');

async function listForTask(taskId) {
  const { rows } = await pool.query(
    'SELECT * FROM activity_log WHERE task_id = $1 ORDER BY created_at DESC LIMIT 50',
    [taskId]
  );
  return rows;
}

module.exports = { listForTask };

File: backend/app/routes/activities.py
"""Activity Routes — Task activity log."""

from fastapi import APIRouter
from ..database import query

router = APIRouter()


@router.get("/{task_id}")
def list_for_task(task_id: str):
    return query(
        "SELECT * FROM activity_log WHERE task_id = %s ORDER BY created_at DESC LIMIT 50",
        (task_id,),
    )
